query/src/main/scala/filodb/query/util/HierarchicalQueryExperience.scala

```diff
@@ -231,16 +234,44 @@ object HierarchicalQueryExperience extends StrictLogging {
       val updatedMetricName = getNextLevelAggregatedMetricName(metricColumnFilter, params, filters)
       updatedMetricName match {
         case Some(metricName) =>
-          val updatedFilters = upsertFilters(filters, Seq(ColumnFilter(metricColumnFilter, Equals(metricName))))
-          logger.info(s"[HierarchicalQueryExperience] Query optimized with filters: ${updatedFilters.toString()}")
-          updatedFilters
+          // Checking if the metric actually ends with the next level aggregation metricSuffix.
+          // If so, update the filters and emit metric
+          // else, return the filters as is
+          metricName.endsWith(params.metricSuffix) match {
```

Comment:
for ExcludeAggRule is the same logic or vice versa

Reply:
Yes, same logic either wise